File: car_data.py
# ──────────────────────────────────────────────
#  ACC car model → class / manufacturer mapping
# ──────────────────────────────────────────────

CAR_MODELS: dict[int, dict] = {
    # ── GT3 ──────────────────────────────────
    0:  {"name": "Porsche 991 GT3 R",            "manufacturer": "Porsche",      "class": "GT3"},
    1:  {"name": "Mercedes AMG GT3",             "manufacturer": "Mercedes",     "class": "GT3"},
    2:  {"name": "Ferrari 488 GT3",              "manufacturer": "Ferrari",      "class": "GT3"},
    3:  {"name": "Audi R8 LMS",                  "manufacturer": "Audi",         "class": "GT3"},
    4:  {"name": "Lamborghini Huracán GT3",       "manufacturer": "Lamborghini",  "class": "GT3"},
    5:  {"name": "McLaren 650S GT3",             "manufacturer": "McLaren",      "class": "GT3"},
    6:  {"name": "Nissan GT-R GT3 2018",         "manufacturer": "Nissan",       "class": "GT3"},
    7:  {"name": "BMW M6 GT3",                   "manufacturer": "BMW",          "class": "GT3"},
    8:  {"name": "Bentley Continental GT3 2018", "manufacturer": "Bentley",      "class": "GT3"},
    9:  {"name": "Porsche 991.2 GT3 Cup",        "manufacturer": "Porsche",      "class": "CUP"},
    10: {"name": "Nissan GT-R GT3 2017",         "manufacturer": "Nissan",       "class": "GT3"},
    11: {"name": "Bentley Continental GT3 2016", "manufacturer": "Bentley",      "class": "GT3"},
    12: {"name": "AMR V12 Vantage GT3",          "manufacturer": "Aston Martin", "class": "GT3"},
    13: {"name": "Reiter Lamborghini G3",        "manufacturer": "Lamborghini",  "class": "GT3"},
    14: {"name": "Emil Frey Jaguar G3",          "manufacturer": "Jaguar",       "class": "GT3"},
    15: {"name": "Emil Frey Jaguar G3 (v2)",     "manufacturer": "Jaguar",       "class": "GT3"},
    16: {"name": "Lexus RC F GT3",               "manufacturer": "Lexus",        "class": "GT3"},
    17: {"name": "Lamborghini Huracán GT3 Evo",  "manufacturer": "Lamborghini",  "class": "GT3"},
    18: {"name": "Honda NSX GT3",                "manufacturer": "Honda",        "class": "GT3"},
    19: {"name": "Lamborghini Huracán ST",       "manufacturer": "Lamborghini",  "class": "ST"},
    20: {"name": "Audi R8 LMS Evo",             "manufacturer": "Audi",         "class": "GT3"},
    21: {"name": "AMR V8 Vantage GT3",           "manufacturer": "Aston Martin", "class": "GT3"},
    22: {"name": "Honda NSX GT3 Evo",            "manufacturer": "Honda",        "class": "GT3"},
    23: {"name": "McLaren 720S GT3",             "manufacturer": "McLaren",      "class": "GT3"},
    24: {"name": "Porsche 991 II GT3 R",         "manufacturer": "Porsche",      "class": "GT3"},
    25: {"name": "Ferrari 488 GT3 Evo",          "manufacturer": "Ferrari",      "class": "GT3"},
    26: {"name": "Mercedes AMG GT3 Evo",         "manufacturer": "Mercedes",     "class": "GT3"},
    27: {"name": "Ferrari 488 Challenge Evo",    "manufacturer": "Ferrari",      "class": "CUP"},
    28: {"name": "BMW M2 CS Racing",             "manufacturer": "BMW",          "class": "TCX"},
    29: {"name": "Porsche 992 GT3 Cup",          "manufacturer": "Porsche",      "class": "CUP"},
    30: {"name": "Lamborghini Huracán ST Evo2",  "manufacturer": "Lamborghini",  "class": "ST"},
    31: {"name": "BMW M4 GT3",                   "manufacturer": "BMW",          "class": "GT3"},
    32: {"name": "Audi R8 LMS GT3 Evo II",      "manufacturer": "Audi",         "class": "GT3"},
    33: {"name": "Ferrari 296 GT3",              "manufacturer": "Ferrari",      "class": "GT3"},
    34: {"name": "Lamborghini Huracán GT3 Evo2", "manufacturer": "Lamborghini",  "class": "GT3"},
    35: {"name": "Porsche 992 GT3 R",            "manufacturer": "Porsche",      "class": "GT3"},
    36: {"name": "McLaren 720S GT3 Evo",         "manufacturer": "McLaren",      "class": "GT3"},
    37: {"name": "Ford Mustang GT3",             "manufacturer": "Ford",         "class": "GT3"},
    # ── GT4 ──────────────────────────────────
    50: {"name": "Alpine A110 GT4",              "manufacturer": "Alpine",       "class": "GT4"},
    51: {"name": "AMR V8 Vantage GT4",           "manufacturer": "Aston Martin", "class": "GT4"},
    52: {"name": "Audi R8 LMS GT4",             "manufacturer": "Audi",         "class": "GT4"},
    53: {"name": "BMW M4 GT4",                   "manufacturer": "BMW",          "class": "GT4"},
    55: {"name": "Chevrolet Camaro GT4R",        "manufacturer": "Chevrolet",    "class": "GT4"},
    56: {"name": "Ginetta G55 GT4",              "manufacturer": "Ginetta",      "class": "GT4"},
    57: {"name": "KTM X-BOW GT4",               "manufacturer": "KTM",          "class": "GT4"},
    58: {"name": "Maserati MC GT4",              "manufacturer": "Maserati",     "class": "GT4"},
    59: {"name": "McLaren 570S GT4",             "manufacturer": "McLaren",      "class": "GT4"},
    60: {"name": "Mercedes AMG GT4",             "manufacturer": "Mercedes",     "class": "GT4"},
    61: {"name": "Porsche 718 Cayman GT4",       "manufacturer": "Porsche",      "class": "GT4"},
    62: {"name": "Toyota GR Supra GT4",          "manufacturer": "Toyota",       "class": "GT4"},
}

# Short abbreviation shown in the overlay logo slot
MANUFACTURER_ABBR: dict[str, str] = {
    "Porsche":      "PORS",
    "Mercedes":     "MERC",
    "Ferrari":      "FERR",
    "Audi":         "AUDI",
    "Lamborghini":  "LAMB",
    "McLaren":      "MCL",
    "Nissan":       "NISS",
    "BMW":          "BMW",
    "Bentley":      "BENT",
    "Aston Martin": "AMR",
    "Jaguar":       "JAG",
    "Lexus":        "LEX",
    "Honda":        "HOND",
    "Alpine":       "ALP",
    "Chevrolet":    "CHEV",
    "Ginetta":      "GIN",
    "KTM":          "KTM",
    "Maserati":     "MAS",
    "Toyota":       "TOY",
    "Ford":         "FORD",
}

# Class display order
CLASS_ORDER = ["GT3", "GT4", "CUP", "ST", "TCX"]

# Class accent colours  (used by web overlay)
CLASS_COLORS: dict[str, str] = {
    "GT3": "#cc0000",
    "GT4": "#cc6600",
    "CUP": "#0055cc",
    "ST":  "#006600",
    "TCX": "#550099",
}


def get_car_info(model_type: int) -> dict:
    return CAR_MODELS.get(
        model_type,
        {"name": f"Car #{model_type}", "manufacturer": "Unknown", "class": "GT3"},
    )


def get_car_class(model_type: int) -> str:
    return get_car_info(model_type)["class"]


def get_manufacturer(model_type: int) -> str:
    return get_car_info(model_type)["manufacturer"]


def get_manufacturer_abbr(model_type: int) -> str:
    mfr = get_manufacturer(model_type)
    return MANUFACTURER_ABBR.get(mfr, mfr[:4].upper())

File: src/car-data.js
'use strict';

// ── ACC car model IDs → class / manufacturer ──────────────────────────────────
//
//  Source: Kunos Simulazioni ACC Broadcast SDK (official)
//
//  IMPORTANT: IDs 0–14 were always correct.  The old table had a duplicate
//  Jaguar at ID 15 (actual = Lexus RC F GT3), which shifted IDs 16–36 by +1.
//  In the GT4 range the old table skipped ID 54 (Chevrolet Camaro GT4.R),
//  shifting IDs 55–61 by +1.  Both are corrected here.
//
const CAR_MODELS = {
  // ── GT3 ──────────────────────────────────────────────────────────────────
  0:  { manufacturer: 'Porsche',      class: 'GT3' },   // 991 GT3 R (2018)
  1:  { manufacturer: 'Mercedes',     class: 'GT3' },   // AMG GT3 (2015)
  2:  { manufacturer: 'Ferrari',      class: 'GT3' },   // 488 GT3 (2018)
  3:  { manufacturer: 'Audi',         class: 'GT3' },   // R8 LMS (2015)
  4:  { manufacturer: 'Lamborghini',  class: 'GT3' },   // Huracán GT3 (2015)
  5:  { manufacturer: 'McLaren',      class: 'GT3' },   // 650S GT3 (2015)
  6:  { manufacturer: 'Nissan',       class: 'GT3' },   // GT-R Nismo GT3 (2018)
  7:  { manufacturer: 'BMW',          class: 'GT3' },   // M6 GT3 (2017)
  8:  { manufacturer: 'Bentley',      class: 'GT3' },   // Continental GT3 (2018)
  9:  { manufacturer: 'Porsche',      class: 'CUP' },   // 991 II GT3 Cup (2017)
  10: { manufacturer: 'Nissan',       class: 'GT3' },   // GT-R Nismo GT3 (2015)
  11: { manufacturer: 'Bentley',      class: 'GT3' },   // Continental GT3 (2015)
  12: { manufacturer: 'Aston Martin', class: 'GT3' },   // V12 Vantage GT3
  13: { manufacturer: 'Lamborghini',  class: 'GT3' },   // Gallardo R-EX (Reiter)
  14: { manufacturer: 'Jaguar',       class: 'GT3' },   // G3 (Emil Frey)
  15: { manufacturer: 'Lexus',        class: 'GT3' },   // RC F GT3            ← was Jaguar (WRONG)
  16: { manufacturer: 'Lamborghini',  class: 'GT3' },   // Huracán GT3 EVO     ← was Lexus
  17: { manufacturer: 'Honda',        class: 'GT3' },   // NSX GT3             ← was Lamborghini
  18: { manufacturer: 'Lamborghini',  class: 'ST'  },   // Huracán ST          ← was Honda
  19: { manufacturer: 'Audi',         class: 'GT3' },   // R8 LMS EVO          ← was Lamborghini ST
  20: { manufacturer: 'Aston Martin', class: 'GT3' },   // V8 Vantage GT3      ← was Audi
  21: { manufacturer: 'Honda',        class: 'GT3' },   // NSX GT3 EVO         ← was Aston Martin
  22: { manufacturer: 'McLaren',      class: 'GT3' },   // 720S GT3            ← was Honda
  23: { manufacturer: 'Porsche',      class: 'CUP' },   // GT2 RS CS Evo       ← was McLaren GT3
  24: { manufacturer: 'Ferrari',      class: 'GT3' },   // 488 GT3 EVO 2020    ← was Porsche GT3
  25: { manufacturer: 'Mercedes',     class: 'GT3' },   // AMG GT3 EVO 2020    ← was Ferrari GT3
  26: { manufacturer: 'Ferrari',      class: 'CUP' },   // 488 Challenge Evo   ← was Mercedes GT3
  27: { manufacturer: 'BMW',          class: 'TCX' },   // M2 CS Racing        ← was Ferrari CUP
  28: { manufacturer: 'Porsche',      class: 'CUP' },   // 911 GT3 Cup 2021    ← was BMW TCX
  29: { manufacturer: 'Lamborghini',  class: 'ST'  },   // Huracán ST EVO2     ← was Porsche CUP
  30: { manufacturer: 'BMW',          class: 'GT3' },   // M4 GT3              ← was Lamborghini ST
  31: { manufacturer: 'Audi',         class: 'GT3' },   // R8 LMS EVO II       ← was BMW GT3
  32: { manufacturer: 'Ferrari',      class: 'GT3' },   // 296 GT3             ← was Audi GT3
  33: { manufacturer: 'Lamborghini',  class: 'GT3' },   // Huracán GT3 EVO2    ← was Ferrari GT3
  34: { manufacturer: 'Porsche',      class: 'GT3' },   // 992 GT3 R           ← was Lamborghini GT3
  35: { manufacturer: 'McLaren',      class: 'GT3' },   // 720S GT3 EVO        ← was Porsche GT3
  36: { manufacturer: 'Ford',         class: 'GT3' },   // Mustang GT3         ← was McLaren GT3
  // ID 37 removed (phantom entry from the old shifted table)

  // ── GT4 ──────────────────────────────────────────────────────────────────
  50: { manufacturer: 'Alpine',       class: 'GT4' },   // A110 GT4
  51: { manufacturer: 'Aston Martin', class: 'GT4' },   // Vantage AMR GT4
  52: { manufacturer: 'Audi',         class: 'GT4' },   // R8 LMS GT4
  53: { manufacturer: 'BMW',          class: 'GT4' },   // M4 GT4
  54: { manufacturer: 'Chevrolet',    class: 'GT4' },   // Camaro GT4.R        ← was missing!
  55: { manufacturer: 'Ginetta',      class: 'GT4' },   // G55 GT4             ← was Chevrolet
  56: { manufacturer: 'KTM',          class: 'GT4' },   // X-Bow GT4           ← was Ginetta
  57: { manufacturer: 'Maserati',     class: 'GT4' },   // GranTurismo MC GT4  ← was KTM
  58: { manufacturer: 'McLaren',      class: 'GT4' },   // 570S GT4            ← was Maserati
  59: { manufacturer: 'Mercedes',     class: 'GT4' },   // AMG GT4             ← was McLaren
  60: { manufacturer: 'Porsche',      class: 'GT4' },   // 718 Cayman GT4 CS   ← was Mercedes
  61: { manufacturer: 'Toyota',       class: 'GT4' },   // GR Supra GT4        ← was Porsche
  // ID 62 removed (Toyota was shifted here by the missing-54 bug; correct ID is 61)
};

const MFR_ABBR = {
  'Porsche':      'PORS', 'Mercedes': 'MERC', 'Ferrari':      'FERR',
  'Audi':         'AUDI', 'Lamborghini': 'LAMB', 'McLaren':   'MCL',
  'Nissan':       'NISS', 'BMW':      'BMW',  'Bentley':      'BENT',
  'Aston Martin': 'AMR',  'Jaguar':   'JAG',  'Lexus':        'LEX',
  'Honda':        'HOND', 'Alpine':   'ALP',  'Chevrolet':    'CHEV',
  'Ginetta':      'GIN',  'KTM':      'KTM',  'Maserati':     'MAS',
  'Toyota':       'TOY',  'Ford':     'FORD',
};

const CLASS_ORDER  = ['GT3', 'GT4', 'CUP', 'ST', 'TCX'];
const CLASS_COLORS = {
  GT3: '#eb0000', GT4: '#eb7900', CUP: '#0055cc', ST: '#006600', TCX: '#0089eb',
};

function getCarInfo(modelType) {
  return CAR_MODELS[modelType] ?? { manufacturer: 'Unknown', class: 'GT3' };
}
function getCarClass(modelType)        { return getCarInfo(modelType).class; }
function getManufacturer(modelType)    { return getCarInfo(modelType).manufacturer; }
function getManufacturerAbbr(modelType) {
  const m = getManufacturer(modelType);
  return MFR_ABBR[m] ?? m.slice(0, 4).toUpperCase();
}

module.exports = { CLASS_ORDER, CLASS_COLORS, getCarClass, getManufacturerAbbr };
